assets/src/application/ui/input/InlineEditInput/Editable.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { Typography } from 'antd';
+import classNames from 'classnames';
+
+import './style.css';
+import { EditableProps } from './types';
+
+const { Text, Title, Paragraph } = Typography;
+
+const EditableTypes = {
+	text: Text,
+	heading: Title,
+	textarea: Paragraph,
+};
+
+const Editable = ({ inputType, onChange, ellipsis, className, ...rest }: EditableProps) => {
+	const Component = EditableTypes[inputType];
+
+	const htmlClasses = classNames('ee-inline-edit', `ee-inline-edit-${inputType}`, className);
```

Comment:
`className` ?

Reply:
That is a prop and should not be mutated :)

Reply:
understood, in this case I would go with:
```
const Editable = ({ inputType, onChange, ellipsis, ...rest }: EditableProps) => {
```
and
```
const className = classNames('ee-inline-edit', `ee-inline-edit-${inputType}`, rest.className);

```